sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -168,7 +168,6 @@ void gnrc_ipv6_demux(kernel_pid_t iface, gnrc_pktsnip_t *current, gnrc_pktsnip_t
         case PROTNUM_ICMPV6:
             DEBUG("ipv6: handle ICMPv6 packet (nh = %u)\n", nh);
             gnrc_icmpv6_demux(iface, pkt);
-            gnrc_pktbuf_release(pkt);
```

Comment:
This removal was necessary because there isn't the release we had in https://github.com/RIOT-OS/RIOT/pull/5525/files#diff-9af59eae807169797112fdc7b1ca9504L140. Additionally, this simplifies a lot since we return to the event loop here anyways.